### PR TITLE
Evaluations for the trajectory including APE and RPE

### PR DESCRIPTION
**Pull request by Chensiyu00**
The pull request first introduces two parts in pypose direction.

1. geometry.py
    In this part, a function named **svdstf()** is introduced.  Different from **svdtf()**, **svdstf()** can get the Sim(3) optimization to align two sets of points.

2. metric.py
    The basic process for evaluating the APE is: 
           1. Synchronize the estimated trajectories using the timestamps, 
           2. Align two trajectories using SE(3) or Sim(3) or None which is according to the setting given by users, 
           3. Calculate the errors ('translation', 'rotation' and etc.) based on requirements
    
     The basic process for evaluating the RPE is: 
           1. Synchronize the estimated trajectories using the timestamps, 
           2. Align two trajectories using SE(3) or Sim(3) or None which is according to the setting given by users,
           3. Get the pose pairs to calculate based on frame or translation.
           4. Calculate the errors ('translation', 'rotation' and etc.) based on requirements

The robustness evalution will be realized later.

### CODE REVIEW

File: pypose/function/geometry.py [hunk 1]
```diff
@@ -311,6 +311,79 @@ def knn(ref, nbr, k=1, ord=2, dim=-1, largest=False, sorted=True):
     dist = torch.linalg.norm(diff, dim=dim, ord=ord)
     return dist.topk(k, dim=dim, largest=largest, sorted=sorted)
 
+def svdstf(source: torch.Tensor, target: torch.Tensor, with_scale: bool=True):
```
Comment (ericzzj1989) [inline pypose/function/geometry.py:314]: viewed

**Review by wang-chen — CHANGES_REQUESTED** [review]

File: tests/function/test_metric.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import torch
+import numpy as np
+import pypose as pp
+from pypose.function.metric import compute_APE, compute_RPE, StampedSE3
```
[the remark this answers is not in the document]
Comment (ericzzj1989) [reply tests/function/test_metric.py:4]: Change `pypose.function.metric.compute_APE, compute_RPE, StampedSE3` to `pypose.function.ape` and `pypose.function.rpe`, not `pypose.metric.ape` and `pypose.metric.rpe`

**Review by wang-chen — APPROVED** [review]
